chain/src/sapling.rs

```diff
@@ -1,38 +1,81 @@
 use std::fmt;
 use hex::ToHex;
 
+/// 
 #[derive(Clone)]
 pub struct Sapling {
-	pub amount: i64,
+	/// The net value of Spend transfers minus Output transfers in a transaction is
+	/// called the balancing_value.
+	/// A positive balancing_value takes value from the Sapling value pool and adds
+	/// it to the transparent value pool.
+	/// A negative balancing value does the reverse.
+	pub balancing_value: i64,
+	/// A Spend transfer spends a note. Its Spend description includes a Pedersen value
+	/// commitment to the value of the note. It is associated with an instance of a Spend
+	/// statement for which it provides a zk-SNARK proof.
 	pub spends: Vec<SaplingSpendDescription>,
+	/// An Output transfer creates a note. Its Output description includes a Pedersen value
+	/// commitment to the note value. It is associated with an instance of an Output statement
+	/// for which it provides a zk-SNARK proof.
 	pub outputs: Vec<SaplingOutputDescription>,
+	/// Consistency of balancing_value with the value commitments in Spend descriptions
```

Comment:
If you want new line in generated html (which I hope you do)

it has to be declared as empty doc comment lines afaik, like

```rust
	/// This signature has a dual role in Sapling:
    ///
 	/// 1) to prove that the total value spent by Spend transfers, minus that produced by
 	///    Output transfers, is consistent with the v balance field of the transaction;
    ///
 	/// 2) To prove that the signer knew the randomness used for the spend and output value
 	///    commitments, in order to prevent Output descriptions from being replayed by an
 	///    adversary in a different transaction. (A Spend description already cannot be
 	///    replayed due to its spend authorization signature.)
```